activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -540,8 +540,7 @@ def disable_referential_integrity
       end
 
       # Override to check all foreign key constraints in a database.
-      def all_foreign_keys_valid?
```

Comment:
`all_foreign_keys_valid?()` is one of the public API https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/AbstractAdapter.html#method-i-all_foreign_keys_valid-3F so we are not able to remove it without any deprecation warnings.

Please refer to https://api.rubyonrails.org/classes/Module.html#method-i-deprecate to make this method deprecated and add some tests to make sure this method is deprecated https://api.rubyonrails.org/classes/ActiveSupport/Testing/Deprecation.html#method-i-assert_deprecated